src/ergw_aaa_diameter_client.erl

```diff
@@ -0,0 +1,142 @@
+%% Copyright 2017, Travelping GmbH <info@travelping.com>
+
+%% This program is free software; you can redistribute it and/or
+%% modify it under the terms of the GNU General Public License
+%% as published by the Free Software Foundation; either version
+%% 2 of the License, or (at your option) any later version.
+
+-module(ergw_aaa_diameter_client).
+
+-include_lib("kernel/include/inet.hrl").
+
+-include_lib("diameter/include/diameter.hrl").
+
+-include("include/diameter_nasreq_rfc7155.hrl").
+
+-export([init/1,
+         stop/0,
+         call/1,
+         cast/1]).
+
+%% diameter callbacks
+-export([peer_up/3,
+         peer_down/3,
+         pick_peer/4,
+         prepare_request/3,
+         prepare_retransmit/3,
+         handle_answer/4,
+         handle_error/4,
+         handle_request/3]).
+
+%%===================================================================
+%% API
+%%===================================================================
+
+init(Opts) ->
+    OriginHost = proplists:get_value(origin_host, Opts, <<"client.example.com">>),
+    OriginRealm = proplists:get_value(origin_realm, Opts, <<"example.com">>),
+    SvcOpts = [{'Origin-Host', OriginHost},
+               {'Origin-Realm', OriginRealm},
+               {'Origin-State-Id', diameter:origin_state_id()},
+               {'Vendor-Id', 0},
```

Comment:
Do we know which L4 protocol needs to be supported? I discusses with @vances that many providers and vendors don't even support SCTP for Diameter communication